smarts/core/sumo_traffic_simulation.py

```diff
@@ -931,14 +938,10 @@ def _reroute_vehicles(self, vehicle_states):
             self._traci_conn.vehicle.setRoute(vehicle_id, new_route_edges)
 
     def _route_for_vehicle(self, vehicle_id: str) -> Optional[List[str]]:
-        if not self.connected:
-            return None
-        try:
-            route = self._traci_conn.vehicle.getRoute(vehicle_id)
```

Comment:
A note on `self._traci_conn.vehicle.getRoute(vehicle_id)`

The TraCI `getRoute` method returns edge ids which form the vehicle's route, but any junctions in the route are not included. Hence, total route distance calculation using `getRoute` method results in a shorter distance.

Reply:
In the public facing method we should be resolving it to a full route using the road map.

Reply:
The following original issues appears to be resolved by this PR.
- Issue-1: In all remaining steps within that episode, both SumoTrafficSimulation and LocalTrafficProvider appear to be managing "Leader-007".
- Issue-2: The destination road of "Leader-007" appears to have changed from E4 under SumoTrafficSimulation control to E2.224.85 when controlled by LocalTrafficProvider.


Below we use the public methods of SumoTrafficSimulation to compute the route distance.
```python
route = traffic_sim.route_for_vehicle(vehicle_name)
for r in route.road_ids:
    print(r)
dist_tot = route.road_length
```
- Prior to this PR, the public method `route_for_vehicle` had internally called `self._traci_conn.vehicle.getRoute(vehicle_id)` which resulted in it returning a shorter route distance. 
- Now, including code changes in this PR, the public methods above still appear to return shorter route distance and does not include junctions in the route.

Reply:
Interesting, I was under the assumption that `RoadMap.route_from_road_ids(road_ids)` would resolve the full route.

Reply:
Tracing the call chain, we see roads added without resolving junctions.
`SumoTrafficSimulation.route_for_vehicle()` -> `SumoRoadNetwork.route_from_road_ids()` -> `SumoRoadNetwork.Route.from_road_ids()`

https://github.com/huawei-noah/SMARTS/blob/23bb3958663bdbfd42e473f5b45f8da0bf26b038/smarts/core/route_cache.py#L70-L80

Similarly, junctions appear to be excluded resulting in a shorter distance in the public method of Smarts traffic provider `LocalTrafficProvider.route_for_vehicle(vehicle_name)`.

We may merge this PR, and track this issue in a separate thread.

Reply:
This may still result in a shorter length perhaps but the junctions will be included.

Reply:
The fixed `SumoTrafficSimulation.route_for_vehicle().road_length` now returns the full length of all the roads (including junctions) in the route, ignoring the starting offset.

Reply:
There should be some consideration in the future if to add the initial and end offsets as necessary attributes of a route.